### PR TITLE
Lazy sqlalchemy

### PR DESCRIPTION
**Pull request by carmelp16**

### CODE REVIEW

File: anyway/utilities.py [hunk 1]
```diff
@@ -197,17 +197,28 @@ def delete_all_rows_from_table(conn, table):
...
+        select = select.where(column_to_chunk_by < next_val)
+    cursor = conn.execution_options(stream_results=True).execute(select)
+    while chunk := cursor.fetchmany(chunk_size):
         logging.debug("after running query on chunk")
         yield [dict(row.items()) for row in chunk]
+
+def split_query_to_chunks_by_column(base_select, column_to_chunk_by, chunk_size, conn):
+    fetch_generator = fetch_first_and_every_nth_value_for_column(conn, column_to_chunk_by, chunk_size)
+    val = next(fetch_generator, None)
+    while next_val := next(fetch_generator, None):
```
Comment (tkalir) [inline anyway/utilities.py:212]: maybe changing the while to "while val", moving the "next = " inside the loop, and dropping 216-220 will work?

File: anyway/utilities.py [hunk 2]
```diff
@@ -197,17 +197,28 @@ def delete_all_rows_from_table(conn, table):
...
-
-def split_query_to_chunks_by_column(base_select, column_to_chunk_by, chunk_size, conn):
-    column_values = fetch_first_and_every_nth_value_for_column(conn, column_to_chunk_by, chunk_size)
-    logging.debug("after fetching every nth column")
-    for index in range(len(column_values)):
-        select = base_select.where(column_to_chunk_by >= column_values[index])
-        if index + 1 < len(column_values):
-            select = select.where(column_to_chunk_by < column_values[index + 1])
-        chunk = conn.execute(select).fetchall()
+def _yield_select_in_chunks(base_select: str, column_to_chunk_by: str, val: typing.Any, next_val: typing.Any, conn, chunk_size: int):
```
Comment (tkalir) [inline anyway/utilities.py:200]: base_select is not str, as can be seen in 201. I believe it is sqlalchemy.sql.expression.Select
also conn is sqlalchemy.engine.Connection

File: anyway/utilities.py [hunk 3]
```diff
@@ -180,9 +180,9 @@ def fetch_first_and_every_nth_value_for_column(conn, column_to_fetch, n):
     select_query = select([sub_query]) \
         .where(or_(func.mod(sub_query.c.row_number, n) == 0,
                    sub_query.c.row_number == 1))
-    ids_and_row_numbers = conn.execute(select_query).fetchall()
-    ids = [id_and_row_number[0] for id_and_row_number in ids_and_row_numbers]
-    return ids
+    cursor = conn.execution_options(stream_results=True).execute(select_query)
```
Comment (tkalir) [inline anyway/utilities.py:183]: from what I understand from the docs, stream_results=True will help only if we use psycopg2, is that what we use? 

https://docs.sqlalchemy.org/en/13/core/connections.html

Comment (carmelp16) [reply anyway/utilities.py:183]: Yes 👍🏽

File: anyway/utilities.py (hunk 2 above)
Comment (carmelp16) [reply anyway/utilities.py:200]: Added 😄

File: anyway/utilities.py (hunk 1 above)
Comment (carmelp16) [reply anyway/utilities.py:212]: Fixed 🥳

**Review by atalyaalon — APPROVED** [review]
@carmelp16 approved, please make sure data load still works as expected (that all rows are loaded etc)

**Review by ziv17 — COMMENTED** [review]
Very nice!

**Comment by ziv17** [comment]
Do we still need chunks if we can read lazily from the database?

**Pull request merged**